### PR TITLE
Ignore empty fields indexer

### PR DESCRIPTION
**Pull request by jnioche**
The indexers currently generate fields where the value can be empty. This PR introduces a new configuration

_indexer.ignore.empty.fields_ with a default value of _false_.

If set to true, the indexing classes will not generate a field if its value is empty or null.

### CODE REVIEW

File: external/elasticsearch/src/main/java/com/digitalpebble/stormcrawler/elasticsearch/bolt/IndexerBolt.java [hunk 1]
```diff
@@ -184,8 +184,10 @@ public void execute(Tuple tuple) {
 
             // display text of the document?
             if (StringUtils.isNotBlank(fieldNameForText())) {
-                String text = tuple.getStringByField("text");
-                builder.field(fieldNameForText(), trimText(text));
+                final String text = trimText(tuple.getStringByField("text"));
+                if (!ignoreEmptyFields() || StringUtils.isNotBlank(text)) {
```
[the remark this answers is not in the document]
Comment (jnioche) [reply external/elasticsearch/src/main/java/com/digitalpebble/stormcrawler/elasticsearch/bolt/IndexerBolt.java:188]: @sonatype-lift ignore

**Comment by jnioche** [comment]
@rzo1 @sebastian-nagel do you think the default value should be true?

File: external/elasticsearch/src/main/java/com/digitalpebble/stormcrawler/elasticsearch/bolt/IndexerBolt.java [hunk 2]
```diff
@@ -199,6 +201,9 @@ public void execute(Tuple tuple) {
             for (String fieldName : keyVals.keySet()) {
                 String[] values = keyVals.get(fieldName);
                 if (values.length == 1) {
+                    if (!ignoreEmptyFields() || StringUtils.isNotBlank(values[0])) {
```
Comment (rzo1) [inline external/elasticsearch/src/main/java/com/digitalpebble/stormcrawler/elasticsearch/bolt/IndexerBolt.java:204]: Isn't it added twice? Line 205 and 207 (if the if is true)

**Review by rzo1 — COMMENTED** [review]
It would be consistent with current behaviour if it remains `false`. However,  I don't have a strong opinion here.

**Comment by jnioche** [comment]
thanks a lot @rzo1

**Pull request merged**